### ISSUE TITLE
Panic through delta after `bat cache --build` after adding a syntax file

### ISSUE THREAD

**Issue by onox** [sharkdp/bat#2449]
**What steps will reproduce the bug?**

1. Install [delta](https://github.com/dandavison/delta) 0.15.1
2. Store https://github.com/wiremoons/ada-sublime-syntax/blob/main/ada-sublime-syntax/Ada.sublime-syntax in `.config/bat/syntaxes/`. (There are no other files are or folders in `.config/bat/`) (I know this file has already been added to recent version of bat, but I'm on an ancient Ubuntu 22.04 distro)
3. Run `bat cache --build`
4. In `~/.gitconfig` have the following content:

```ini
[core]
    pager = delta
[interactive]
    diffFilter = delta --color-only
[delta]
    side-by-side = true
    line-numbers = true
[merge]
    conflictstyle = diff3
[diff]
    colorMoved = default
```
5. Create git repo with following `main.adb` file:

```ada
with Ada.Text_IO;

procedure Main is
begin
   Ada.Text_IO.Put_Line ("Hello world");
end Main;
```

6. `git commit` the file and then change the text a bit.

7. Run `git diff`.

**What happens?**

`git diff` prints following file:

```
[bat warning]: Unknown theme 'Monokai Extended', using default.
thread 'main' panicked at 'something is very wrong if the default theme is missing', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/bat-0.21.0/src/assets.rs:203:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

`bat --list-themes` and `bat --list-languages` work fine, including the theme Monokai Extended.

**What did you expect to happen instead?**

Printed diff.

**Workaround**

1. Run `rm ~/.cache/bat/themes.bin`
2. `git diff` shows printed diff, but without syntax highlighting

**Additional info**

`bat main.adb` shows syntax highlighting on .adb files if `~/.cache/bat/themes.bin` is present, otherwise there is no highlighting. But with the `themes.bin` file present, `bat` seems to panic when invoked through `delta`.

**How did you install `bat`?**

Using `apt` on Ubuntu 22.04.

Bat version: 0.19.0-1ubuntu0.1

**Comment by keith-hall** [comment]
Is it possible that the `bat` version you have installed is using a different/incompatible asset binary format than the version of `delta` you are using? Being on an ancient distro shouldn't mean you can't use the latest version of `bat`...

**Comment by onox** [comment]
> is using a different/incompatible asset binary format

I'm not sure what you mean. How can I determine this?

**Comment by keith-hall** [comment]
Well, you said you're using the latest version of `delta`, released on December 3rd last year, and `bat`  `0.19`. In `bat`'s changelog we see that for version `v0.21.0`, released May 12 last year, there was a breaking change to the file format of `syntaxes.bin`. So as your `bat` version is older than that, then the binary format it is using is incompatible with the format `delta` is expecting.
